packages/upload/src/FzUpload.vue

```diff
@@ -93,11 +99,16 @@ function handleDrop(event: DragEvent) {
 function handleInputChange(event: Event) {
   if (!input.value?.files) return;
   addFiles([...input.value.files]);
+  emit("fzupload:change", model.value);
   input.value.value = "";
 }
 
 function addFiles(filesToAdd: File[]) {
-  model.value = [...(model.value ?? []), ...filesToAdd];
+  if(props.multiple) {
+    model.value = [...(model.value ?? []), ...filesToAdd];
+  } else {
+    model.value = [...filesToAdd];
```

Comment:
It's not granted that you only have one file. If it comes from the input file then yes it should. But if it's come drag and drop, you have to handle that yourself.